### ISSUE TITLE
Cannot convert the "c50a4bf0-db51-4ad2-88d5-fe2287da0eb8" value of type "Deserialized.Microsoft.Dynamics.Nav.Apps.Types.AppId" to type "System.Guid"

### ISSUE THREAD

**Issue by jwikman** [microsoft/navcontainerhelper#3374]
**Describe the issue**
Starting this week we get this error in our NextMajor pipelines, one week ago all was fine.
```
Cannot bind parameter 'Id'. Cannot convert the "c50a4bf0-db51-4ad2-88d5-fe2287da0eb8" value of type "Deserialized.Microsoft.Dynamics.Nav.Apps.Types.AppId" to type "System.Guid".
```
Since this error was thrown inside a dynamically created .ps1 (caused by `Get-BcContainerAppInfo`) file that was executed in the container, it was bit a challenging to hunt down.

This line in the script in the container causes this error
```
$apps = Get-NAVAppInfo @inArgs | Where-Object { (!$installedOnly) -or ($_.IsInstalled -eq $true) } | ForEach-Object { Get-NAVAppInfo -id $_.AppId -publisher $_.publisher -name $_.name -version $_.Version @inArgs }
```
with $inArgs being:
```
Name                           Value
----                           -----
ServerInstance                 BC
Tenant                         default
TenantSpecificProperties       False
```

These two lines (executed in the container) reproduces the issue
```
$apps = Get-NAVAppInfo @inArgs
Get-NAVAppInfo -id $apps[0].AppId @inArgs
```
While this works:
```
$apps = Get-NAVAppInfo @inArgs
Get-NAVAppInfo -id $apps[0].AppId.Value @inArgs
```
It seems like the result from `Get-NAVAppInfo` has changed from previous versions. The Id is not a guid or string, but the type `Microsoft.Dynamics.Nav.Apps.Types.AppId`.

**Full output of scripts**
```
BcContainerHelper version 6.0.6
Version: 24.0.16561.0
```

**Additional context**
- does it happen all the time
  - Yes, for NextMajor containers. Starting sometime in the last week
- did it use to work?
  - Yes, one week ago (BC v24.0.16336.0)

**Comment by freddydk** [comment]
Please try BcContainerHelper preview - should be fixed there

**Comment by jwikman** [comment]
Ok, thanks!

**Comment by jwikman** [comment]
6.0.7-preview1150 did indeed solve this issue.

**Comment by salgiza** [comment]
While I can also confirm that the preview did solve the problem for us, wouldn't it be better if the ticket stays open until a stable release with the solution is available?

**Comment by jwikman** [comment]
Sure, I was just happy that I solved the issues in our pipeline. I cannot wait on the stable release with this... :)

Anyway, reopened as requested!

**Comment by rvanbekkum** [comment]
Even with the work-around in place, we keep getting the following error for NextMajor build pipeline runs (see screenshot): "The tenant default is not accessible"

![image](https://github.com/microsoft/navcontainerhelper/assets/7383241/bd5f3ebc-c68e-4cf1-923c-ee1a1ca02b9c)

We have a Retry Count on the task and it only fails the first attempt, while the second attempt it works just fine.

This only happens with the NextMajor (24.x) build pipeline runs, not with builds against the stable releases (23.x).

Anyone else experiencing these issues and/or know what to do about it?

**Comment by freddydk** [comment]
@rvanbekkum your problem looks to be different - maybe caused by something else.
please create a new issue with this and repro steps for this.
In 24.x, all CmdLets are now running PowerShell 7 and use a bridge from PS5.

**Comment by freddydk** [comment]
BCContainerHelper 6.0.7 has shipped with a fix for this (with generic images 1.0.2.15 or above)
Also note that generic images 1.0.2.16 for the latest Windows versions has shipped.
